src/test/java/com/snowflake/kafka/connector/internal/streaming/TopicPartitionChannelTest.java

```diff
@@ -1245,4 +1247,68 @@ public void testOffsetTokenVerificationFunction() {
     Assert.assertFalse(StreamingUtils.offsetTokenVerificationFunction.verify("1", "3", "4", 3));
     Assert.assertFalse(StreamingUtils.offsetTokenVerificationFunction.verify("2", "1", "4", 3));
   }
+
+  @Test
+  public void assignANewChannelAfterTheSetupIsFullyDone() throws Exception {
+    // given
+
+    SnowflakeStreamingIngestChannel channel1 = Mockito.mock(SnowflakeStreamingIngestChannel.class);
```

Comment:
Nit: I would like to use `FakeSnowflakeStreamingIngestChannel` or extend that implementation so we don't have to do so much mocking.

Reply:
Either way, I must modify `FakeSnowflakeStreamingIngestChannel` for a specific edge case. This means it won't be significantly different from mocking an object.